lab_2_retrieval_w_bm25/main.py

```diff
@@ -138,6 +282,39 @@ def rank_documents(
 
     In case of corrupt input arguments, None is returned.
     """
+    if (not isinstance(indexes, list)
+        or not all(isinstance(dict_in_indexes, dict) for dict_in_indexes in indexes)
+        or not isinstance(stopwords, list)
+        or not all(isinstance(word, str) for word in stopwords)
+            or not isinstance(query, str)):
+        return None
+    for dict_in_indexes in indexes:
+        for key, value in dict_in_indexes.items():
+            if not isinstance(key, str) or not isinstance(value, float):
+                return None
+
+    tokenized_query = tokenize(query)
+    if not isinstance(tokenized_query, list) or not stopwords:
+        return None
+    tokenized_query_without_stopwords = remove_stopwords(tokenized_query, stopwords)
+    if not isinstance(tokenized_query_without_stopwords, list):
+        return None
+    l_of_tuples_dict_plus_its_relevance = []
+
+    for dict_index, dict_at_the_index in enumerate(indexes):
+        relevance_of_the_particular_doc = 0.0
+        for word in tokenized_query_without_stopwords:
+            if word in dict_at_the_index:
+                relevance_of_the_particular_doc += dict_at_the_index[word]
```

Comment:
can we also use sum() as in `calculate_idf` function?

Reply:
done!